### PR TITLE
Sync less project attributes. 

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Remove 500 single-item sync calls from oop to our host for these attributes.  Prior to the change we had:

image

These represent number of calls where we're asking for a single object of that type.  

Post the change we have:

image

This means we only make 3 calls from oop to host for this data.  One for 314 items, one for 1 item, and one for 270 items.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Moving to draft.  i want to collect current before/after numbers on this before proceeding.

File: src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs [hunk 1]
```diff
@@ -270,10 +270,13 @@ public static SolutionStateChecksums Deserialize(ObjectReader reader)
...
 
-                if (projectState.TryGetStateChecksums(out var projectStateChecksums) &&
-                    searchingChecksumsLeft.Remove(projectStateChecksums.Checksum))
+                if (projectState.TryGetStateChecksums(out var projectStateChecksums))
                 {
-                    result[projectStateChecksums.Checksum] = projectStateChecksums;
+                    if (searchingChecksumsLeft.Remove(projectStateChecksums.Checksum))
+                        result[projectStateChecksums.Checksum] = projectStateChecksums;
+
+                    if (searchingChecksumsLeft.Remove(projectStateChecksums.Info))
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs:278]: >if (searchingChecksumsLeft.Remove(projectStateChecksums.Info)) [](http://example.com/codeflow?start=20&length=62)

This feels like a bug fix, but I'm curious whether it's an optimization and I'm missing something

Comment (ToddGrun) [reply src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs:278]: Oh, I see. The new GetAssetsAsync call you added for the optimization uses the Info with the TopLevelProjects path.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski 	For review when you get back.

**Pull request merged**